examples/test_util/EclIO.cpp

```diff
@@ -0,0 +1,756 @@
+/*
+   Copyright 2019 Equinor ASA.
+
+   This file is part of the Open Porous Media project (OPM).
+
+   OPM is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   OPM is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+   */
+
+#include "EclIO.hpp"
+
+#include <algorithm>
+#include <iterator>
+#include <iomanip>
+#include <stdexcept>
+#include <typeinfo>
+
+
+namespace Opm {
+  namespace ecl {
+    namespace io {
+
+
+
+bool isEOF(std::fstream *fileH){
+    
+    /* try to read 4 byte (integer), if passed eof of, return true.
+       if not, seek back to current file possition  */
+    
+    int num;
+    long int pos=fileH->tellg();
+    fileH->read((char *)&num, sizeof(num));
+    
+    if (fileH->eof()){
+        return true;
+    } else {
+        fileH->seekg (pos);
+        return false;
+    }
+}
+
+template<>
+auto flipEndian<int>(const int &num) {
+
+   unsigned int tmp=__builtin_bswap32(num);
+   
+   return (int)tmp;
+}
+
+template<>
+auto flipEndian<float>(const float &num) {
+
+    float value=num;
+
+    char *floatToConvert = reinterpret_cast<char*>(&value);
+    std::reverse(floatToConvert, floatToConvert+4);
+    
+    return value;
+}
+
+template<>
+auto flipEndian<double>(const double &num) {
+    
+    double value=num;
+    
+    char *doubleToConvert = reinterpret_cast<char*>(&value);
+    std::reverse(doubleToConvert, doubleToConvert+8);
+
+    return value;
+}
+
+std::tuple<const int, const int> block_size_data(EIOD::eclArrType arrType){
+
+    switch(arrType) {
+        case EIOD::INTE  : return std::make_tuple(EIOD::sizeOfInte,EIOD::MaxBlockSizeInte);   break;
+        case EIOD::REAL  : return std::make_tuple(EIOD::sizeOfReal,EIOD::MaxBlockSizeReal);   break;
+        case EIOD::DOUB  : return std::make_tuple(EIOD::sizeOfDoub,EIOD::MaxBlockSizeDoub);   break;
+        case EIOD::LOGI  : return std::make_tuple(EIOD::sizeOfLogi,EIOD::MaxBlockSizeLogi);   break;
+        case EIOD::CHAR  : return std::make_tuple(EIOD::sizeOfChar,EIOD::MaxBlockSizeChar);   break;
+        case EIOD::MESS  : throw std::invalid_argument("Type 'MESS' have not assosiated data") ;   break;
+    }
+}
+
+std::string trimr(const std::string &str1) {
+
+    std::string tmpStr=str1;
+    
+    while (tmpStr.back()==' '){
+        tmpStr.pop_back();
+    }
+
+    return std::move(tmpStr);
+}
+
+
+void readBinaryHeader(std::fstream &fileH,std::string &arrName,int &size,EIOD::eclArrType &arrType){
```

Comment:
No big deal - but a detail for the future: I think most people; myself included prefer a space following the `,` - i.e.
```
void readBinaryHeader(std::fstream &fileH, std::string& arrName, int& size, ...)
```